src/Nox.Reference.Data.World/Models/TaxNumbers/TaxNumberValidationResult.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Text;
+
+namespace Nox.Reference;
+
+public class TaxNumberValidationResult
+{
+    private readonly List<string?> _validationErrors = new();
+
+    private TaxNumberValidationResult()
+    {
+    }
+
+    public ValidationStatus Status { get; private set; } = ValidationStatus.Unverified;
+    public object? ApiVerificationData { get; set; }
+    public IReadOnlyList<string?> ValidationErrors => _validationErrors;
+    public string FormattedTaxNumber { get; init; } = string.Empty;
+    public string OriginalTaxNumber { get; init; } = string.Empty;
+    public string Country { get; init; } = string.Empty;
+
+    public static TaxNumberValidationResult CreateWithoutValidation(string validationError)
+    {
+        var result = new TaxNumberValidationResult();
+        if (!string.IsNullOrWhiteSpace(validationError))
+        {
+            result.AddError(validationError);
+        }
+
+        result.Status = ValidationStatus.Unverified;
+        return result;
+    }
+
+    public static TaxNumberValidationResult CreateWithValidation(string taxNumber, string country)
+    {
+        var result = new TaxNumberValidationResult
+        {
+            OriginalTaxNumber = taxNumber,
+            FormattedTaxNumber = NormalizeTaxNumber(taxNumber, country),
+            Country = country,
+            Status = ValidationStatus.Valid
+        };
+        return result;
+    }
+
+    public void AddError(string error)
+    {
+        if (error != null)
+        {
+            Status = ValidationStatus.Invalid;
+        }
+        _validationErrors.Add(error);
+    }
+
+    public void AddErrors(IEnumerable<string> errors)
```

Comment:
result.AddErrors(ChecksumValidationHelper.ValidateLuhnDigitNumber(digitPart));

Looks like we expect error here. Might well method name be changed. 
AddErrors -> HandleValidation() or Check() etc..

Reply:
Also, maybe signature can be changed Func<string, string>() validationFn ?

Reply:
Renamed